app/jobs/dspace_publication_job.rb

```diff
@@ -1,26 +1,43 @@
 class DspacePublicationJob < ActiveJob::Base
   queue_as :default
 
-  def perform(thesis, sqs_client)
+  def setup(sqs_client)
     # note, we pass in the sqs_client to allow for easier stubbing even though it is inconvenient for non-test
     # purposes
-    queue_url = ENV.fetch('SQS_INPUT_QUEUE_URL')
+    @queue_url = ENV.fetch('SQS_INPUT_QUEUE_URL')
+    Rails.logger.info("Sending a message to queue #{@queue_url}...")
 
-    Rails.logger.info("Sending a message to queue #{queue_url}...")
+    # We allow the SQS client to be passed in to allow for easier mocking in tests. However, passing the client in via
+    # the controller in dev/prod leads to errors. This is non-ideal, but it allow for us to use a client if one is
+    # passed and create one if not.
+    @sqs_client = sqs_client
+    @sqs_client ||= Aws::SQS::Client.new(region: ENV.fetch('AWS_REGION'))
```

Comment:
TIL the `||=` syntax

Would this be less prone to errors if the method tested that the provided client was valid? Or is that just inviting more problems?

Reply:
I'll think about checking the client. I'm currently thinking we may not need to check it as I think in reality we'll only pass in a client from the tests and that's a mocked client. In real usage I think we'll always create one here.